stetho/src/main/java/com/facebook/stetho/inspector/elements/android/ViewDescriptor.java

```diff
@@ -10,19 +10,71 @@
 package com.facebook.stetho.inspector.elements.android;
 
 import android.view.View;
+import android.view.ViewDebug;
+import com.facebook.stetho.common.Accumulator;
+import com.facebook.stetho.common.LogUtil;
 import com.facebook.stetho.common.StringUtil;
 import com.facebook.stetho.common.android.ResourcesUtil;
-import com.facebook.stetho.inspector.elements.AttributeAccumulator;
 import com.facebook.stetho.inspector.elements.AbstractChainedDescriptor;
+import com.facebook.stetho.inspector.elements.AttributeAccumulator;
+import com.facebook.stetho.inspector.elements.StyleInfo;
 
 import javax.annotation.Nullable;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
 import java.util.Map;
+import java.util.Set;
 
 final class ViewDescriptor extends AbstractChainedDescriptor<View> implements HighlightableDescriptor {
   private static final String ID_ATTRIBUTE_NAME = "id";
+  private static final Set<Method> viewGetters;
 
   private final MethodInvoker mMethodInvoker;
 
+  static {
```

Comment:
I see 2 performance problems with this approach. Since this code is executing in the static initializer for ViewDescriptor, it will always run when the Elements tab is booted up even if the CSS stuff isn't being used. That violates pay-for-play, which is very important for a project like Stetho. The second problem is due to this being stored in static fields, which means it is not eligible for garbage collection once all Stetho clients have disconnected.

Because `ViewDescriptor` is essentially a singleton, it is perfectly valid for us to store this type of data in instance fields (btw, non-final/mutable static fields should be prefixed with 's'. final, or rather "effectively const" fields, should be ALL_UPPERCASE). And, this data should not be initialized until it's needed ... which looks like it'd be the first call to `onGetStyles()`. A good idea would be to hide this behind a method that handles the deferred initialization, e.g.

```
/**
  *  NOTE: Only access this via {@link #getViewGetters}.
  */
@GuardedBy("this")
private volatile Set<Method> mViewGetters;

private Set<Method> getViewGetters() {
  if (mViewGetters == null) {
    synchronized (this) {
      ... initialize mViewGetters ...
    }
  }
  return mViewGetters;
}

@Override
protected void onGetStyles(...) {
  for (Method method : getViewGetters()) { 
    ...
  }
}
```